src/panels/lovelace/editor/view-editor/hui-view-visibility-editor.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  html,
+  LitElement,
+  TemplateResult,
+  customElement,
+  property,
+  PropertyValues,
+  CSSResult,
+  css,
+} from "lit-element";
+import "@polymer/paper-input/paper-input";
+
+import { HomeAssistant } from "../../../../types";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { configElementStyle } from "../config-elements/config-elements-style";
+import { LovelaceViewConfig, ShowViewConfig } from "../../../../data/lovelace";
+
+import { fetchUsers, User } from "../../../../data/user";
+import memoizeOne from "memoize-one";
+import { compare } from "../../../../common/string/compare";
+import { HaSwitch } from "../../../../components/ha-switch";
+
+declare global {
+  interface HASSDomEvents {
+    "view-visibility-changed": {
+      visible: ShowViewConfig[];
+    };
+  }
+}
+
+@customElement("hui-view-visibility-editor")
+export class HuiViewVisibilityEditor extends LitElement {
+  set config(config: LovelaceViewConfig) {
+    this._config = config;
+    this._visible = this._config.visible;
+  }
+
+  @property() public hass!: HomeAssistant;
+  @property() public _config!: LovelaceViewConfig;
+  @property() private _users!: User[];
+  @property() private _visible?: boolean | ShowViewConfig[];
+
+  private _sortedUsers = memoizeOne((users: User[]) => {
+    return users
+      .filter((user) => !user.system_generated)
+      .sort((a, b) => compare(a.name, b.name));
+  });
+
+  protected firstUpdated(changedProps: PropertyValues) {
+    super.firstUpdated(changedProps);
+
+    fetchUsers(this.hass).then((users) => {
+      this._users = users;
+      // how can I resize and reposition dialog? Should I use `iron-resize` event?
+      // fireEvent(this as HTMLElement, "iron-resize");
+    });
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass || !this._users) {
+      return html``;
+    }
+
+    return html`
+      ${configElementStyle}
+      <div class="card-config">
+        <span>Select which users should have access to this view</span>
+
+        ${this._sortedUsers(this._users).map(
+          (user) => html`
+            <div class="flex">
+              <div>${user.name}</div>
+              <ha-switch
+                .userId="${user.id}"
+                @change=${this.valChange}
+                .checked=${this.checkUser(user.id)}
+                >&nbsp;</ha-switch
+              >
+            </div>
+          `
+        )}
+      </div>
+    `;
+  }
+
+  protected checkUser(userId: string): boolean {
+    if (this._visible === undefined) {
+      return true;
+    }
+    if (typeof this._visible === "boolean") {
+      return this._visible as boolean;
+    }
+    return (this._visible as ShowViewConfig[]).some((u) => u.user === userId);
+  }
+
+  private valChange(ev: Event): void {
+    const userId = (ev.currentTarget as any).userId;
+    const checked = (ev.currentTarget as HaSwitch).checked;
+
+    if (this._visible === undefined) {
+      this._visible = true;
+    }
```

Comment:
Why do this if we are going to change it anyway?

Reply:
Ok, then I'll check  `_visible` property to be required - `@property() private _visible!: boolean | ShowViewConfig[];`

Reply:
No, it can be undefined, so that is not right. Just add it to the check below.

Reply:
https://github.com/home-assistant/frontend/pull/4978/files#diff-33e1deebcb55da6fbd461e09dbe65d67R35
this prevents it from being undefined.
It should in theory :)

Reply:
No, you set it to true if it is undefined, so it was undefined.

Reply:
In recent commit I've changed:
`@property() private _visible?: boolean | ShowViewConfig[];`
to
`@property() private _visible!: boolean | ShowViewConfig[];`
and I'm setting `visible` like this:
```
  set config(config: LovelaceViewConfig) {
    this._config = config;
    if (this._config.visible === undefined) {
      this._config.visible = true;
    } else {
      this._visible = this._config.visible;
    }
  }
```

this should work fine in my opinion

Reply:
`this._config.visible = true;` should be `this._visible = this._config.visible;`?

Reply:
you are right. I fixed that. I also changed logic in `valChange` I'm not creating a copy of array when adding new users, instead, I use array.push